mkt/site/tests/test_views.py

```diff
@@ -66,6 +66,13 @@ def test_404_api(self):
         res = self.client.get('/api/this-should-never-work/')
         eq_(res.status_code, 404)
         eq_(res.content, '')
+        self.assertCORS(res, 'get')
+
+    def test_404_api_debug(self):
+        with self.settings(DEBUG=True):
+            res = self.client.get('/api/this-should-never-work/')
+            eq_(res.status_code, 404)
+            self.assertCORS(res, 'get')
```

Comment:
Yes but making an actual options request like ``self.client.options(...)`` can execute completely different code yielding different response headers so it's worth adding that request in a test. For example, DRF won't execute the original view at all when you do an options request so sometimes it won't hit any decorators like you'd expect.